index.js

```diff
@@ -827,7 +827,11 @@
 
   //  Array$prototype$reduce :: Array a ~> ((b, a) -> b, b) -> b
   function Array$prototype$reduce(f, initial) {
-    return this.reduce(function(acc, x) { return f(acc, x); }, initial);
+    var acc = initial;
+    for (var i = 0; i < this.length; i += 1) {
+      acc = f(acc, this[i]);
+    }
+    return acc;
   }
```

Comment:
Does this change make an appreciable difference?

Reply:
I've measured the difference in the [benchmark suite for List](https://funkia.github.io/list/benchmarks/). I'd say the difference is quite significant.

![screenshot_20180319_142156](https://user-images.githubusercontent.com/521604/37597848-d286faac-2b80-11e8-946d-15973681a475.png)

Reply:
Significant indeed!

Now that I know we'll keep the code I'll share my formatting request. I'd like to see this:

```javascript
for (var idx = 0; idx < this.length; idx += 1) acc = f(acc, this[idx]);
```

Reply:
:zap: